tgui/packages/tgui/interfaces/ListInput.js

```diff
@@ -0,0 +1,173 @@
+    message,
+    buttons,
+    timeout,
+  } = data;
+
+  // Search
+  const [showSearchBar, setShowSearchBar] = useLocalState(context, "search_bar", false);
+  const [displayedArray, setDisplayedArray] = useLocalState(context, "displayed_array", buttons);
+
+  // KeyPress
+  const [searchArray, setSearchArray] = useLocalState(context, "search_array", []);
+  const [searchIndex, setSearchIndex] = useLocalState(context, "search_index", 0);
+  const [lastCharCode, setLastCharCode] = useLocalState(context, "last_char_code", null);
+
+  // Selected Button
+  const [selectedButton, setSelectedButton] = useLocalState(context, "selected_button", null);
+  return (
+    <Window
+      title={title}
+      width={325}
+      height={325}
+    >
+      {timeout !== undefined && <Loader value={timeout} />}
+      <Window.Content>
+        <Section fill>
+          <Flex
+            height={showSearchBar? "80%" : "90%"}
+          >
+            <Section
+              scrollable
+              width="100%"
+              fill
+              title={message}
+              tabIndex={0}
+              onKeyDown={e => {
+                const charCode = String.fromCharCode(e.keyCode);
+                if (!charCode || lastScrollTime > performance.now()) return;
+                lastScrollTime = performance.now() + 150;
+
+                let foundValue;
+                if (charCode === lastCharCode && searchArray.length > 0) {
+                  const nextIndex = searchIndex + 1;
+
+                  if (nextIndex < searchArray.length) {
+                    foundValue = searchArray[nextIndex];
+                    setSearchIndex(nextIndex);
+                  }
+                  else {
+                    foundValue = searchArray[0];
+                    setSearchIndex(0);
+                  }
+                }
+                else {
+                  const resultArray = displayedArray.filter(value =>
+                    value.substring(0, 1) === String.fromCharCode(e.keyCode)
+                  );
+
+                  if (resultArray.length > 0) {
+                    setSearchArray(resultArray);
+                    setSearchIndex(0);
+                    foundValue = resultArray[0];
+                  }
+                }
+
+                if (foundValue) {
+                  setLastCharCode(charCode);
+                  setSelectedButton(foundValue);
+                  document.getElementById(foundValue).focus();
+                }
+              }}
+              buttons={(
+                <Button
+                  icon="search"
+                  color="transparent"
+                  selected={showSearchBar}
+                  tooltip="Search Bar"
+                  tooltipPosition="left"
+                  onClick={() => {
+                    setShowSearchBar(!showSearchBar);
+                    setDisplayedArray(buttons);
+                  }}
+                  compact
+                />
+              )}
+              level={2}
```

Comment:
Why is this level 2, make it simple.

Reply:
Hard to do this because I need the section for the scrollbar specifically so that you don't, for example, need to scroll to the bottom to access the searchbar or press the `Confirm` or `Cancel` buttons. They'd need to be separate from the section with the scrollbar, or new code would need to be written.